compiler/testData/diagnostics/tests/EnumEntryAsType.fir.kt

```diff
@@ -6,49 +6,49 @@ enum class Color {
     }
 }
 
-class MyColor(val x: <!UNRESOLVED_REFERENCE!>Color.RED<!>, y: <!UNRESOLVED_REFERENCE!>Color.RED<!>) : <!UNRESOLVED_REFERENCE!>Color.RED<!> {
+class MyColor(val x: Color.<!ENUM_ENTRY_AS_TYPE!>RED<!>, y: Color.<!ENUM_ENTRY_AS_TYPE!>RED<!>) : Color.<!ENUM_ENTRY_AS_TYPE!>RED<!> {
 
-    var z: <!UNRESOLVED_REFERENCE!>Color.RED<!> = Color.RED
-    set(arg: <!UNRESOLVED_REFERENCE!>Color.RED<!>) { z = arg }
+    var z: Color.<!ENUM_ENTRY_AS_TYPE!>RED<!> = Color.RED
+    set(arg: Color.<!ENUM_ENTRY_AS_TYPE!>RED<!>) { z = arg }
 
-    fun foo(arg: <!UNRESOLVED_REFERENCE!>Color.RED<!>): <!UNRESOLVED_REFERENCE!>Color.RED<!> = arg
+    fun foo(arg: Color.<!ENUM_ENTRY_AS_TYPE!>RED<!>): Color.<!ENUM_ENTRY_AS_TYPE!>RED<!> = arg
 
-    fun bar(): <!UNRESOLVED_REFERENCE!>Color.RED<!> {
-        class Local : <!UNRESOLVED_REFERENCE!>Color.RED<!>
-        fun local(arg: <!UNRESOLVED_REFERENCE!>Color.RED<!>): <!UNRESOLVED_REFERENCE!>Color.RED<!> = arg
-        val temp: <!UNRESOLVED_REFERENCE!>Color.RED<!> = Color.RED
-        temp as? <!UNRESOLVED_REFERENCE, UNRESOLVED_REFERENCE!>Color.RED<!>
-        if (temp is <!UNRESOLVED_REFERENCE!>Color.RED<!>) {
-        return temp as <!UNRESOLVED_REFERENCE!>Color.RED<!>
+    fun bar(): Color.<!ENUM_ENTRY_AS_TYPE!>RED<!> {
+        class Local : Color.<!ENUM_ENTRY_AS_TYPE!>RED<!>
+        fun local(arg: Color.<!ENUM_ENTRY_AS_TYPE!>RED<!>): Color.<!ENUM_ENTRY_AS_TYPE!>RED<!> = arg
+        val temp: Color.<!ENUM_ENTRY_AS_TYPE!>RED<!> = Color.RED
+        temp as? Color.<!ENUM_ENTRY_AS_TYPE, ENUM_ENTRY_AS_TYPE!>RED<!>
```

Comment:
Any idea why `ENUM_ENTRY_AS_TYPE` appears twice here? Probably the same reason `UNRESOLVED_REFERENCE` appeared twice before.

Reply:
Thanks! It turns out to be a bug in the FIR tree. Filed https://youtrack.jetbrains.com/issue/KT-47291 tracking it.